### DISCUSSION TITLE
SDKs and Examples

### DISCUSSION THREAD

**Discussion by hifabienne** [zitadel/zitadel#7376] [Polls]
We have started with the most used languages and frameworks and now we want to know what you would like to see.

Find the list of current SDKs and Examples here:
https://zitadel.com/docs/sdk-examples/introduction

**Comment by snird** [comment]
Since RoR is already in the lead, may I suggest the obvious library to implement it with? https://github.com/omniauth/omniauth-openid

**Comment by danielloader** [comment]
I'd like to vote for [Envoy ext_authz](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/ext_authz_filter) as a long shot request - given it's available in the following (and presumably others) implementations:

- https://istio.io/latest/docs/tasks/security/authorization/authz-custom/
- https://www.envoyproxy.io/docs/envoy/v1.30.1/start/sandboxes/ext_authz.html
- https://gateway.envoyproxy.io/v1.0.1/tasks/security/ext-auth/ 
- https://docs.solo.io/gloo-mesh-enterprise/latest/security/external-auth/server/
- https://developer.hashicorp.com/consul/docs/connect/proxies/envoy-extensions/configuration/ext-authz
- https://www.getambassador.io/docs/edge-stack/latest/topics/running/services/ext-authz
- https://projectcontour.io/docs/v1.10.0/config/client-authorization/
- https://github.com/cilium/cilium/issues/29831
- https://docs.konghq.com/mesh/latest/features/meshopa/
- https://www.alibabacloud.com/help/en/asm/user-guide/implement-custom-authorization-by-using-the-grpc-protocol
- https://cloud.google.com/apigee/docs/api-platform/envoy-adapter/v2.0.x/operation#multi-tenant-environment-support

Having a working example of how one could use Envoy's ext_authz protocol inline on a reverse proxy/ingress/gateway/service mesh would be a huge benefit.

I appreciate this actually might require extra software engineering, as I'm unsure how one would bridge the Authorizations/Grants API Zitadel exposes and how you'd use it to check if you could POST/DELETE to an /example path or simply GET.

It's entirely possible the authz side of the equation will have to live inside the application code using the SDKs but thought I'd throw it out there.

As a simpler request, Istio's Request Authorization path could be documented too and how to set up Istio to simply check your JWT is valid at the edge of your service mesh or between micro services if they pass the token down the call chain:

- https://istio.io/latest/docs/reference/config/security/request_authentication/

**Comment by nileshtrivedi** [comment]
Other options I would like to see:

- FastAPI (Python)
- SvelteKit (Javascript)
- Phoenix framework (Elixir)

**Reply by dasrocky** [reply]
I would love some examples for Fastapi too.

**Reply by oskar-gmerek** [reply]
> * SvelteKit (Javascript)

⊕ for SvelteKit (Typescript)

**Reply by Dargun** [reply]
what is with c++

**Comment by soernt-poppe** [comment]
* Flutter
* C#

**Comment by Vipulkushwaha** [comment]
Can we have an updated SvelteKit 5 example? It seems like it is getting complex to implement seeing the recent SvelteKit developments.

**Reply by Vipulkushwaha** [reply]
Added #9555

**Comment by guedressel** [comment]
Please also consider PHP Symfony as it is a popular and mature framework other tools build on.
https://symfony.com/doc/current/security.html
